beacon_node/beacon_chain/src/schema_change/migration_schema_v24.rs

```diff
@@ -0,0 +1,292 @@
+                                        "error computing previous_state_root {e:?}"
+                                    ))
+                                })?
+                        };
+
+                        let diff_base_state_root =
+                            if let Some(diff_base_slot) = storage_strategy.diff_base_slot() {
+                                DiffBaseStateRoot::new(
+                                    diff_base_slot,
+                                    state_summaries_dag
+                                        .ancestor_state_root_at_slot(state_root, diff_base_slot)
+                                        .map_err(|e| {
+                                            Error::MigrationError(format!(
+                                                "error computing ancestor_state_root_at_slot {e:?}"
+                                            ))
+                                        })?,
+                                )
+                            } else {
+                                DiffBaseStateRoot::zero()
+                            };
+
+                        let new_summary = HotStateSummary {
+                            slot,
+                            latest_block_root: old_summary.latest_block_root,
+                            latest_block_slot: old_summary.latest_block_slot,
+                            previous_state_root,
+                            diff_base_state_root,
+                        };
+                        let op = new_summary.as_kv_store_op(state_root);
+                        // It's not ncessary to immediately commit the summaries of states that are
+                        // ReplayFrom. However we do so for simplicity.
+                        db.hot_db.do_atomically(vec![op])?;
+                    }
+                }
+            }
+
+            // 3. Stage old data for deletion.
+            if slot % T::EthSpec::slots_per_epoch() == 0 {
+                migrate_ops.push(KeyValueStoreOp::DeleteKey(
+                    DBColumn::BeaconState,
+                    state_root.as_slice().to_vec(),
+                ));
+            }
+
+            // Delete previous summaries
+            migrate_ops.push(KeyValueStoreOp::DeleteKey(
+                DBColumn::BeaconStateSummary,
+                state_root.as_slice().to_vec(),
+            ));
+
+            summaries_written += 1;
+            if last_log_time.elapsed() > Duration::from_secs(5) {
+                last_log_time = Instant::now();
+                info!(
+                    diffs_written = diffs_written,
+                    summaries_written = summaries_written,
+                    summaries_count = state_summaries_dag.summaries_count(),
+                    "Hot states migration in progress"
+                );
+            }
+        }
+    }
+
+    // TODO(hdiff): Should run hot DB compaction after deleting potentially a lot of states. Or should wait
+    // for the next finality event?
```

Comment:
Yeah I think we should do column-specific compaction after the migration completes (in `schema_change.rs`). I hope it should be reasonably fast because the deleted columns should now be empty.